src/mare/compiler/refer.cr

```diff
@@ -102,6 +102,15 @@ module Mare::Compiler::Refer
       @analysis[node] = info
     end
 
+    def touch(ctx, node : AST::Prefix)
+      case node.op.value
+      when "address_of"
+        unless node.term.is_a? AST::Identifier
```

Comment:
I'm thinking that we need a different check for this:
```suggestion
        unless @analysis[node.term].is_a? Local
```

An identifier could be many different kinds of things, such as a type name for example.

We want to check there that it is specifically a local variable.

Reply:
Sure